### DISCUSSION TITLE
Breaking Changes (Angular)

### DISCUSSION THREAD

**Discussion by Wayne-Mather** [salesforce/akita#786] [General]
Hi,

I don't understand your breaking change and am a little confused if I need to call destroy() to stop memory leaks (unsubscribe) or not.

Your example says this: 

```
@Injectable()
class TodosStore {
  constructor(private store: TodosStore) {}
 
  ngOnDestroy() {
    this.store.destroy();
  }
}
```

This is not component code, and your injecting the store into the store.... If we use providers in our modules or within our components for the stores, do we need to manually call destroy() to stop any memory leaks? 

For example, lets say I have a lazy loaded module and a root component for that called a shell, do we have to do the following for the 2 examples? Please not this is not 100% accurate code as we are talking concept.

```
// store is a provider on the component
@Component({providers: MyStore, MyQuery})
export class MyShell implements OnInit, OnDestory {
 constructor(private store: MyStore, private query: MyQuery) {}
  ngOnInit(): void { // so something }
  ngOnDestroy(): void { this.store.destroy(); }
}
```

```
// store is a provider on the module
@Component()
export class MyShell implements OnInit, OnDestory {
 constructor(private store: MyStore, private query: MyQuery) {}
  ngOnInit(): void { // so something }
  ngOnDestroy(): void { this.store.destroy(); }
}
```

**Comment by NetanelBasal** [comment]
This is a mistake: It should be:

```ts
@Injectable()
class TodosStore extends Store { 
  ngOnDestroy() {
    this.destroy();
  }
}
```

**Reply by Wayne-Mather** [reply]
Thanks, I was totally down the wrong track then - the destroy is just in our stores - understand now.

I wonder why no one else picked up the mistake when starting to look at migrating across to 7? We are still 6.2.3 but have just moved to NG13 and want to upgrade to v7 of Akita as well.

**Reply by NetanelBasal** [reply]
I guess they understood the mistake :)

**Reply by Wayne-Mather** [reply]
Can you also confirm, it's only if you inject via the component, module injection does not need to do this?

**Reply by NetanelBasal** [reply]
Only if you destroy it manually.

**Reply by Wayne-Mather** [reply]
Ok, super clear now - so most of us don't have to do a thing if we use standard DI and Angular component lifecycles. 

Thanks for the clarity and sorry if maybe I am not smart enough to have read all that between the lines 😊
